mempool/auction_bid.go

```diff
@@ -0,0 +1,128 @@
+package mempool
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/auth/signing"
+)
+
+type (
+	// AuctionBidInfo defines the information about a bid to the auction house.
+	AuctionBidInfo struct {
+		Bidder       sdk.AccAddress
+		Bid          sdk.Coin
+		Transactions [][]byte
+		Timeout      uint64
+		Signers      []map[string]struct{}
+	}
+
+	// AuctionBid defines the interface for processing auction transactions. It is
+	// a wrapper around all of the functionality that each application chain must implement
+	// in order for auction processing to work.
+	AuctionBid interface {
```

Comment:
Thought about this some more...what about `AuctionFactory`?

Reply:
Big factory pattern enthusiast. 

Decoupling auction bid from mempool also makes sense to me.

Reply:
Down for `AuctionFactory`. I have some idea's on how we could still use the `ExtensionOptions` on `sdk.Tx` to completely decouple the bid from the mempool. If we only have it store the priority and validate that the bid and priority within the tx and extension option match, then the mempool doesn't need to have conception of what the underlying tx is. This might mess with general UX tho since most wallets and users dont add extension options to begin with. We should maybe explore this approach in the `BlockBuster` refactor.

Reply:
Thinking extension over msg interface?

Reply:
We're still evaluating what the best approach is in the context of the refactor we're planning but msg interface is probably going to be our approach. We'll share the RFC of the refactor with you in the coming days (that'll address the coupling concerns you've been discussing).

Reply:
Definitely going to be curious to hear your thoughts on design and such since the refactor will be a new approach to general mempool design.